### ISSUE TITLE
Show console output in popup when pre-commit hooks fired

### ISSUE THREAD

**Issue by pumano** [desktop/desktop#15286]
### The feature request

Show console output in popup when pre-commit hooks fired, because if that process is long running - it's confuse from UX point of view. First idea - github desktop freezes

### Proposed solution

show hooks console output inside modal as it show final result of failed pre-commit.

**Comment by steveward** [comment]
Thanks for the issue @pumano, dropping https://github.com/desktop/desktop/issues/9351 in here to reference as well since that's related to pre-commit hook failure context.

**Comment by steveward** [comment]
Requested in https://github.com/desktop/desktop/issues/14323:

> The feature request
> I am currently checking type issues, test coverage and other trivial things in the pre-push hook. This is because these issues are identifiable in the local machine and once fixed provides high assurance that the build will be stable.
> 
> Proposed solution
> Step 1: Check if any hooks are configured for the given repo. (The files under .git/hooks/ without .sample extension)
> Step 2: If present, show a console window in GitHub Desktop (Please refer the screen shot - Inline with Commit message but in No changes window - Bottom section)
> Step 3: The output of the respective hook should be dumped here on respective action.

**Comment by KrystilizeNevaDies** [comment]
I would also find this very useful.
I wanted to add that interaction with any popup console for hooks that read cli input would also be much appreciated <3.

**Comment by shamilovtim** [comment]
Yeah to back this up. When working with git in a substantial project with CI, CD and CLI scripts there are many useful messages echoed to the terminal from various git hooks. Without seeing these messages Github Desktop lacks context and becomes a hamstrung client. displaying the echoed messages somewhere is critical and should be a first class piece of any git GUI.

**Comment by steveward** [comment]
From https://github.com/desktop/desktop/issues/16369:

> When cherry-picking a commit into a branch with conflicts that fails Git hooks (in my case, [husky](https://github.com/typicode/husky)), GitHub Desktop switches back to the prior branch with no changes (or errors).
> 
> I have uploaded a video for additional context. If you watch this very closely and very carefully, you can see for a split second there is an error modal that dismisses immediately.

**Comment by NitishGadangi** [comment]
@steveward  It's been a while since this issue opened, is there any scope for bringing this feature soon?
Would be really helpful.

**Comment by ekimbernow** [comment]
+1 for this issue. We are implementing pre-commit hooks and if many files are changed (which sometimes happens), the "Committing to ..." indicator spins for a long time.

Our hooks output messages for files that fail the check and can also provide interactive status (i.e., progress bar) when the number of files is greater than say 100.

**Comment by choxnox** [comment]
Any updates on this issue?

**Comment by ahmad-punch** [comment]
any update here?

When I push with github desktop while running a pre-push hook to check if build succeeds or not
It seems that the github desktop is stuck. (because of no visual indication)

Also I get no output or logs in VSCODE in the git output.

**Comment by steveward** [comment]
Requested in https://github.com/desktop/desktop/issues/19001.

**Comment by niik** [comment]
Yesterday we shipped GitHub Desktop 3.5.5-beta1 to our [beta channel](https://desktop.github.com/beta/) with [improved support for hooks related to committing](https://github.com/desktop/desktop/pull/21319).

<img width="280" alt="Screenshot showing the GitHub Desktop commit form as a pre-commit hook is running" src="https://github.com/user-attachments/assets/b389d45d-90d5-4660-9bab-f9b6b64c8201" />

The support is experimental and currently limited to our beta channel. While we believe that this release resolved this issue we're looking for real-world feedback so if you have the ability we'd appreciate it if you could download our [beta channel](https://desktop.github.com/beta/) and give it a spin. Ensure the experimental hooks support it's enabled by going to Settings -> Git -> Hooks and making sure the "Load Git hook environment variables from shell" is ticked.

<img width="614" height="558" alt="Screenshot showing the hook settings dialog" src="https://github.com/user-attachments/assets/4ed1157c-1c89-4733-88db-dd8fe84cf58f" />

**Comment by ekimbernow** [comment]
Seems to work just as I would like:

<img width="1532" height="911" alt="Image" src="https://github.com/user-attachments/assets/fadacc94-2fac-4819-a5a7-12ba98b51247" />

It would be nice if the window could be resized to accommodate the width of the messages, but what's there is good enough.
